File: src/zh_log.py
# -*- coding: utf-8 -*-
"""中文运行日志：标明当前跑到哪一步、哪里出问题。"""
from __future__ import annotations

import sys
from typing import Any


def _ensure_utf8_stdio() -> None:
    for stream in (sys.stdout, sys.stderr):
        try:
            stream.reconfigure(encoding="utf-8", errors="replace")  # type: ignore[attr-defined]
        except Exception:
            pass


_ensure_utf8_stdio()


def step(module: str, msg: str, **extra: Any) -> None:
    _print(f"[进行中][{module}] {msg}{_fmt(extra)}")


def info(module: str, msg: str, **extra: Any) -> None:
    _print(f"[信息][{module}] {msg}{_fmt(extra)}")


def ok(module: str, msg: str, **extra: Any) -> None:
    _print(f"[完成][{module}] {msg}{_fmt(extra)}")


def fail(module: str, msg: str, **extra: Any) -> None:
    _print(f"[失败][{module}] 失败于：{msg}{_fmt(extra)}", err=True)


def warn(module: str, msg: str, **extra: Any) -> None:
    _print(f"[警告][{module}] {msg}{_fmt(extra)}", err=True)


def _fmt(extra: dict[str, Any]) -> str:
    if not extra:
        return ""
    parts = [f"{k}={v}" for k, v in extra.items()]
    return " | " + " ".join(parts)


def _print(line: str, *, err: bool = False) -> None:
    stream = sys.stderr if err else sys.stdout
    try:
        print(line, flush=True, file=stream)
    except UnicodeEncodeError:
        enc = getattr(stream, "encoding", None) or "utf-8"
        stream.buffer.write((line + "\n").encode(enc, errors="replace"))
        stream.buffer.flush()

File: dsh-manhua/src/zh-log.ts
/** 中文运行日志：标明当前跑到哪一步、哪里出问题。 */

function fmt(extra?: Record<string, unknown>): string {
  if (!extra || !Object.keys(extra).length) return "";
  return (
    " | " +
    Object.entries(extra)
      .map(([k, v]) => `${k}=${v}`)
      .join(" ")
  );
}

export function step(module: string, msg: string, extra?: Record<string, unknown>): void {
  console.log(`[进行中][${module}] ${msg}${fmt(extra)}`);
}

export function info(module: string, msg: string, extra?: Record<string, unknown>): void {
  console.log(`[信息][${module}] ${msg}${fmt(extra)}`);
}

export function ok(module: string, msg: string, extra?: Record<string, unknown>): void {
  console.log(`[完成][${module}] ${msg}${fmt(extra)}`);
}

export function fail(module: string, msg: string, extra?: Record<string, unknown>): void {
  console.error(`[失败][${module}] 失败于：${msg}${fmt(extra)}`);
}

export function warn(module: string, msg: string, extra?: Record<string, unknown>): void {
  console.warn(`[警告][${module}] ${msg}${fmt(extra)}`);
}
